aria/storage/modeling/instance_elements.py

```diff
@@ -264,39 +264,24 @@ def node_fk(cls):
         return cls.foreign_key('node', nullable=True)
 
     @declared_attr
-    def relationship_source_fk(cls):
-        return cls.foreign_key('relationship', nullable=True)
-
-    @declared_attr
-    def relationship_target_fk(cls):
+    def relationship_fk(cls):
         return cls.foreign_key('relationship', nullable=True)
 
     # endregion
 
     description = Column(Text)
     type_name = Column(Text)
+    edge = Column(Text)
 
     # region many-to-one relationships
 
     @declared_attr
     def node(cls):
         return cls.many_to_one_relationship('node')
 
-    @property
-    def relationship(self):
-        return self.relationship_source or self.relationship_target
-
-    @declared_attr
-    def relationship_source(cls):
-        return cls.many_to_one_relationship('relationship',
-                                            foreign_key_column='relationship_source_fk',
-                                            backreference='source_interfaces')
-
     @declared_attr
-    def relationship_target(cls):
-        return cls.many_to_one_relationship('relationship',
-                                            foreign_key_column='relationship_target_fk',
-                                            backreference='target_interfaces')
+    def relationship(cls):
```

Comment:
ok, so there's an issue here. the relationship instance would hold a column "interfaces", this does not imply whether those are source or target interfaces. Only after having a look at each interface we can derive this info according to the edge. 

is this the behavior we want? 
I'm not entirely sure but i could be persuaded :)